esx_service/utils/auth.py

```diff
@@ -42,6 +46,67 @@ def get_auth_mgr():
         thread_local._auth_mgr.connect()
     return thread_local._auth_mgr
 
+def get_fallback_tenant():
+    """ 
+        Get __FALLBACK tenant by querying the auth DB.
+        VMs which does not belong to any tenant explicitly will 
+        be assigned to __FALLBACK tenant if __FALLBACK tenant exists
+        Return value:
+        -- error_info: return None on success or error info on failure
+        -- tenant_uuid: return __FALLBACK tenant uuid on success,  
+           return None on failure or __FALLBACK tenant does not exist
+        -- tenant_name: return __FALLBACK tenant name on success,
+           return None on failure or __FALLBACK tenant does not exist
+    """
+    tenant_uuid = None
+    _auth_mgr = get_auth_mgr()
+    try:
+        cur = _auth_mgr.conn.execute(
+            "SELECT id FROM tenants WHERE name = ?",
+            (FALLBACK_TENANT, )
+            )
+        result = cur.fetchone()
+    except sqlite3.Error as e:
+        error_info = "Error {0} when querying from tenants table for tenant {1}".format(e, FALLBACK_TENANT)
+        logging.error(error_info)
+        return str(e), None, None
+    if result:
+        # found __FALLBACK tenant
+        tenant_uuid = result[0]
+        logging.debug("Found __FALLBACK tenant, tenant_uuid %s, tenant_name %s", tenant_uuid, FALLBACK_TENANT)
+        return None, tenant_uuid, FALLBACK_TENANT
+    else:
+        # cannot find __FALLBACK tenant
+        logging.debug(error_code.TENANT_NOT_EXIST.format(FALLBACK_TENANT))
+        return None, None, None
+
+
+def get_fallback_privileges():
```

Comment:
I think we should differentiate the term "privilege" and "right". IIUC, in our service the term "privilege" indicates some rights (create, remove, mount, etc.) on one datastore. One tenant can have multiple "privileges" on different datastores, but there will be one and only one privilege (which includes multiple "rights") on one single datastore.